tests/tests_adventures.py

```diff
@@ -0,0 +1,50 @@
+import os
+import re
+import utils
+import hedy
+import unittest
+
+path = '../coursedata/adventures'
+files = [f for f in os.listdir(path) if os.path.isfile(os.path.join(path, f)) and f.endswith ('.yaml')]
+
+
+class TestsLevel1(unittest.TestCase):
+  level = 1
+
+  def check_code(self, f, level, field_name, code):
+      # We ignore empty code snippets
+      if len(code) == 0:
+          return True
+      try:
+          hedy.transpile(code, int(level))
+      except Exception as E:
+          if E.args[0] != 'Has Blanks':  # code with blanks is ok!
+              return 'Invalid ' + field_name + ' in file ' + f + ', level ' + str(level) + code
+
+      return True
+
+  def test_adventure_snippets(self):
+    for f in files:
+      f = os.path.join (path, f)
+      yaml = utils.load_yaml_uncached (f)
+
+      adventure_fails = []
+
+      for adventure in yaml ['adventures']:
+        for level in yaml ['adventures'] [adventure] ['levels']:
+
+          result = self.check_code (f, level, 'start_code', yaml ['adventures'] [adventure] ['levels'] [level] ['start_code'])
+          if result != True:
+            adventure_fails.append(result)
+
+          # story_text (between triple backticks)
+          splitted_story_text = re.split (r'\s*```\n', yaml ['adventures'] [adventure] ['levels'] [level] ['story_text'])
+          for k, part in enumerate (splitted_story_text):
+            if k % 2 == 0:
+              continue
+            # Only the odd parts of the text are code snippets
```

Comment:
The logic is as follows: the text snippets are surrounded by triple backticks. We split the text by triple backticks (actually, it's whitespace, followed by triple backticks, followed by a newline).
- If there are no triple backticks, then the text is divided into one part (index 0), so there's no text snippets.
- If there's only one code snippet, it will be the second element of the list (index 1), but what goes before and after (indexes 0 and 2) will not be code snippets.
- If there are two code snippets, they will be at indexes 1 and 3.
- This can be expanded inductively as long as each code snippet has its own triple backticks surrounding it, which I believe is the case.

I will add part of this explanation in a comment on the script. Let me know if it's clear.

Reply:
> The logic is as follows: the text snippets are surrounded by triple backticks. We split the text by triple backticks (actually, it's whitespace, followed by triple backticks, followed by a newline).
> 
> * If there are no triple backticks, then the text is divided into one part (index 0), so there's no text snippets.
> * If there's only one code snippet, it will be the second element of the list (index 1), but what goes before and after (indexes 0 and 2) will not be code snippets.
> * If there are two code snippets, they will be at indexes 1 and 3.
> * This can be expanded inductively as long as each code snippet has its own triple backticks surrounding it, which I believe is the case.
> 
> I will add part of this explanation in a comment on the script. Let me know if it's clear.

If we replace the regex by a markdown parsing library then no explanation will be needed :).

Reply:
Haha yeah if we use an md parser this will all be clear enough :)